From 870dcb5c747a1fa2cfaea5f8c126cca3ea160e5f Mon Sep 17 00:00:00 2001
From: Wenping Zhang <wenping.zhang@rock-chips.com>
Date: Fri, 30 Jun 2017 15:28:56 +0800
Subject: [PATCH] ARM: dts: rk322x: move the gpio_keys node from dtsi to
 product dts.

The keys configuration is constantly changed on different products,
so move it to product's dts.

Change-Id: Ifec408a0529a224b8716797a88b97eddeab10fa7
Signed-off-by: Wenping Zhang <wenping.zhang@rock-chips.com>
---
 arch/arm/boot/dts/rk3229-echo-v10.dts | 23 ++++++++++++++
 arch/arm/boot/dts/rk3229-evb.dts      | 24 +++++++++++++++
 arch/arm/boot/dts/rk3229-gva-sdk.dts  | 44 ++++++++++++++++++++++++---
 arch/arm/boot/dts/rk322x-android.dtsi | 26 ----------------
 4 files changed, 87 insertions(+), 30 deletions(-)

diff --git a/arch/arm/boot/dts/rk3229-echo-v10.dts b/arch/arm/boot/dts/rk3229-echo-v10.dts
index a2903ec2ebf3..fccddd6ab775 100644
--- a/arch/arm/boot/dts/rk3229-echo-v10.dts
+++ b/arch/arm/boot/dts/rk3229-echo-v10.dts
@@ -104,6 +104,24 @@
 		WIFI,host_wake_irq = <&gpio0 28 GPIO_ACTIVE_HIGH>;
 		status = "okay";
 	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		autorepeat;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&pwr_key>;
+
+		power_key: power-key {
+			label = "GPIO Key Power";
+			gpios = <&gpio3 23 GPIO_ACTIVE_LOW>;
+			linux,code = <116>;
+			debounce-interval = <100>;
+			wakeup-source;
+		};
+	};
 };
 
 &gpu {
@@ -279,6 +297,11 @@
 		};
 	};
 
+	keys {
+		pwr_key: pwr-key {
+			rockchip,pins = <3 23 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
 };
 
 &sdio {
diff --git a/arch/arm/boot/dts/rk3229-evb.dts b/arch/arm/boot/dts/rk3229-evb.dts
index f137b0d9d429..027558b25771 100644
--- a/arch/arm/boot/dts/rk3229-evb.dts
+++ b/arch/arm/boot/dts/rk3229-evb.dts
@@ -128,6 +128,24 @@
 			regulator-always-on;
 		};
 	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		autorepeat;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&pwr_key>;
+
+		power_key: power-key {
+			label = "GPIO Key Power";
+			gpios = <&gpio3 23 GPIO_ACTIVE_LOW>;
+			linux,code = <116>;
+			debounce-interval = <100>;
+			wakeup-source;
+		};
+	};
 };
 
 &cpu0 {
@@ -169,6 +187,12 @@
 			rockchip,pins = <3 20 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	keys {
+		pwr_key: pwr-key {
+			rockchip,pins = <3 23 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
 };
 
 &pwm1 {
diff --git a/arch/arm/boot/dts/rk3229-gva-sdk.dts b/arch/arm/boot/dts/rk3229-gva-sdk.dts
index 11b82c23a688..34b5717df127 100644
--- a/arch/arm/boot/dts/rk3229-gva-sdk.dts
+++ b/arch/arm/boot/dts/rk3229-gva-sdk.dts
@@ -88,6 +88,38 @@
 		WIFI,host_wake_irq = <&gpio0 28 GPIO_ACTIVE_HIGH>;
 		status = "okay";
 	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		autorepeat;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&pwr_key &bt_key &mute_key>;
+
+		power_key {
+			label = "GPIO Key Power";
+			gpios = <&gpio1 18 GPIO_ACTIVE_HIGH>;
+			linux,code = <116>;
+			debounce-interval = <100>;
+			wakeup-source;
+		};
+
+		bt_key {
+			label = "Blutooth Key";
+			gpios = <&gpio3 31 GPIO_ACTIVE_HIGH>;
+			linux,code = <237>;
+			debounce-interval = <100>;
+		};
+
+		mute_key {
+			label = "Mute Key";
+			gpios = <&gpio3 20 GPIO_ACTIVE_HIGH>;
+			linux,code = <113>;
+			debounce-interval = <100>;
+		};
+	};
 };
 
 &gpu {
@@ -103,10 +135,6 @@
 	status = "disabled";
 };
 
-&power_key {
-	gpios = <&gpio1 18 GPIO_ACTIVE_HIGH>;
-};
-
 &i2c0 {
 	status = "okay";
 
@@ -279,6 +307,14 @@
 		pwr_key: pwr-key {
 			rockchip,pins = <1 18 RK_FUNC_GPIO &pcfg_pull_down>;
 		};
+
+		bt_key: bt-key {
+			rockchip,pins = <3 31 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+
+		mute_key: mute-key {
+			rockchip,pins = <3 20 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
 	};
 
 	pmic {
diff --git a/arch/arm/boot/dts/rk322x-android.dtsi b/arch/arm/boot/dts/rk322x-android.dtsi
index 211fcdd8cf30..8ba6e7987e87 100644
--- a/arch/arm/boot/dts/rk322x-android.dtsi
+++ b/arch/arm/boot/dts/rk322x-android.dtsi
@@ -62,24 +62,6 @@
 		compatible = "arm,psci-1.0";
 		method = "smc";
 	};
-
-	gpio_keys {
-		compatible = "gpio-keys";
-		#address-cells = <1>;
-		#size-cells = <0>;
-		autorepeat;
-
-		pinctrl-names = "default";
-		pinctrl-0 = <&pwr_key>;
-
-		power_key: power-key {
-			label = "GPIO Key Power";
-			gpios = <&gpio3 23 GPIO_ACTIVE_LOW>;
-			linux,code = <116>;
-			debounce-interval = <100>;
-			wakeup-source;
-		};
-	};
 };
 
 &cpu0 {
@@ -145,14 +127,6 @@
 	supports-sdio;
 };
 
-&pinctrl {
-	keys {
-		pwr_key: pwr-key {
-			rockchip,pins = <3 23 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
-};
-
 &tsadc {
 	rockchip,hw-tshut-mode = <0>; /* tshut mode 0:CRU 1:GPIO */
 	status = "okay";
-- 
2.35.3

